### PR TITLE
Add model and test of max31855 Cold-Junction Compensated Thermocouple-to-Digital Converter

### PR DESCRIPTION
**Pull request by SimonWin**
This pull request adds a model of the spi based max31855 Cold-Junction Compensated Thermocouple-to-Digital Converter chip (https://datasheets.maximintegrated.com/en/ds/MAX31855.pdf). It also adds a test for that model.

### CODE REVIEW

File: src/vcml/models/spi/max31855.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    temp_thermalcouple("temp_thermalcouple", 25),
+    temp_internal("temp_internal", 10),
+    fault("fault", false),
+    scv("scv", false),
+    scg("scg", false),
+    oc("oc", false),
+    spi_in("spi_in") {
+    SC_HAS_PROCESS(max31855);
+    SC_METHOD(cs_edge);
+    sensitive << m_cs.default_event();
```
Comment (SimonWin) [inline src/vcml/models/spi/max31855.cpp:108]: Is it intentional for gpio_target_socket that I need to call default_event here? Just doing sensitive << m_cs throws a warning and doesn't call default_event in gpio_target_socket.

Comment (janweinstock) [reply src/vcml/models/spi/max31855.cpp:108]: This should be fixed in the newest version

**Comment by janweinstock** [comment]
Hi Simon!
Thanks for your PR, here are some questions before I can merge:
1. What happens when the processor sets CS and reads SPI in the same quantum, i.e., before the event has triggered?
2. Why is the chip select socket private? I can always access it through `spi_host`
3. Can we avoid having an extra process, event and gpio socket by just sampling the temperature when in `BYTE0` state?
3a. If we really need the CS (maybe the CPU resets it sometimes mid-sequence?) can we avoid the extra process by using `gpio_notify`?

**Comment by SimonWin** [comment]
Hi,

Thanks for your questions. I think I addressed all of them in my new commit.

I made CS public and switched from using a process to using gpio_notify.
 
If you have more suggestions/questions, please let me know.

**Pull request merged**